block-storage/src/main/scala/coop/rchain/blockstorage/finality/LastFinalizedKeyValueStorage.scala

```diff
@@ -1,46 +1,48 @@
 package coop.rchain.blockstorage.finality
 
 import java.nio.file.Path
-
 import cats.syntax.all._
 import cats.effect.{Concurrent, Sync}
-import coop.rchain.blockstorage.dag.codecs.{codecBlockHash, codecSeqNum}
+import coop.rchain.blockstorage.dag.codecs.{codecBlockHash, codecByteString, codecSeqNum}
 import coop.rchain.metrics.Metrics
 import coop.rchain.models.BlockHash.BlockHash
 import coop.rchain.shared.Log
 import coop.rchain.store.{KeyValueStore, KeyValueTypedStore}
 import coop.rchain.shared.syntax._
 import coop.rchain.blockstorage.syntax._
+import coop.rchain.models.Validator.Validator
 
 class LastFinalizedKeyValueStorage[F[_]: Sync] private (
-    lastFinalizedBlockDb: KeyValueTypedStore[F, Int, BlockHash]
+    lastFinalizedBlockDb: KeyValueTypedStore[F, Validator, BlockHash]
 ) extends LastFinalizedStorage[F] {
-  val fixedKey = 1
-  override def put(blockHash: BlockHash): F[Unit] =
-    lastFinalizedBlockDb.put(Seq((fixedKey, blockHash)))
+  override def put(validator: Option[Validator] = None, blockHash: BlockHash): F[Unit] =
+    lastFinalizedBlockDb.put(
+      Seq((validator.getOrElse(LastFinalizedStorage.noValidatorKey), blockHash))
+    )
```

Comment:
I don't think it's needed to extend existing methods to support new key `LastFinalizedStorage.noValidatorKey`.

Example for `put(validator.some, ...)`.
```scala
def put(validator: Validator, blockHash: BlockHash): F[Unit] =
  lastFinalizedBlockDb.put(Seq((validator, blockHash)))
```
Example for `put(none, ...)` (syntax).
```scala
def put(blockHash: BlockHash): F[Unit] =
  lastFinalizedBlockDb.put(Seq((LastFinalizedStorage.noValidatorKey, blockHash)))
```

Reply:
I'm not sure I understand - I've adjusted this to be able to store finalized blocks for different Validators, to replay finalization process. Option here is to also support local view for readonly nodes.

Reply:
In other words, why to decide about `none` case inside `put` when this can be additional (syntax) method. In that way current interface does not need to be changed.

Reply:
but 
```scala
def put(validator: Validator, blockHash: BlockHash): F[Unit] =
  lastFinalizedBlockDb.put(Seq((validator, blockHash)))
```
cannot be syntax - it has to be trait method, because underlying `lastFinalizedBlockDb` requires validator. So in trait methods we have to be to supply validator. So either we need to add second method with validator, of change the only one to support Option for validator. 
Are you saying that the second option is better?

Reply:
@tgrospic hdy think, maybe put lastFinalizedBlocks database inside BlockDagStorage?

Reply:
It makes sense, but it would be nice that changing is done internal to DagStore and (snapshot) read is exposed through DagRepresentation.
Implementation of DagStore can accept finalization KeyValueTypedStore and use setter for multiple kv pairs.